widget/calendar.go

```diff
@@ -0,0 +1,229 @@
+package widget
+
+import (
+	"image/color"
+	"math"
+	"strconv"
+	"strings"
+	"time"
+
+	"fyne.io/fyne/v2"
+	"fyne.io/fyne/v2/canvas"
+	"fyne.io/fyne/v2/container"
+	"fyne.io/fyne/v2/layout"
+	"fyne.io/fyne/v2/theme"
+	"fyne.io/fyne/v2/widget"
+)
+
+// Declare conformity with Layout interface
+var _ fyne.Layout = (*calendarLayout)(nil)
+
+type calendarLayout struct {
+	cellSize float32
+	padding  float32
+}
+
+func newCalendarLayout(c float32, p float32) fyne.Layout {
+	return &calendarLayout{cellSize: c, padding: p}
+}
+
+func (g *calendarLayout) countRows(objects []fyne.CanvasObject) int {
+	count := 0
+	for _, child := range objects {
+		if child.Visible() {
+			count++
+		}
+	}
+
+	return int(math.Ceil(float64(count) / float64(7)))
```

Comment:
You might want to move that 7 out to a global constant like `daysPerWeek` or something along those lines. I know that I told you to avoid global variables but constants are absolutely fine to have global (as they never change their value and improve code readability).